kaldb/src/main/java/com/slack/kaldb/zipkinApi/ZipkinService.java

```diff
@@ -0,0 +1,591 @@
+    localEndpoint.put("serviceName", "backend");
+    localEndpoint.put("ipv4", "192.168.99.1");
+    localEndpoint.put("port", 3306);
+    HashMap<String, Object> remoteEndpoint = new HashMap<>();
+    remoteEndpoint.put("ipv4", "172.19.0.2");
+    remoteEndpoint.put("port", 58648);
+    HashMap<String, String> tags = new HashMap<>();
+    tags.put("http.method", "GET");
+    tags.put("http.path", "/api");
+    Span span1 =
+        new Span(
+            id,
+            traceId,
+            parentId,
+            name,
+            timestamp,
+            duration,
+            kind,
+            localEndpoint,
+            remoteEndpoint,
+            null,
+            tags);
+    //TODO: output List<Span> as JSON instead of Service of List<Span>
+    Service service = new Service(ImmutableList.of(span1));
+    return JsonUtil.writeJsonArray(JsonUtil.writeAsString(service));
+    */
+
+    // TODO: consider how to qualify the query variables
+    String queryString = "serviceName:" + serviceName;
+    KaldbSearch.SearchRequest.Builder searchRequestBuilder = KaldbSearch.SearchRequest.newBuilder();
+    KaldbSearch.SearchResult searchResult =
+        searcher.doSearch(
+            searchRequestBuilder
+                .setIndexName("testindex") // [Q] as of now we don't need to worry about index?
+                .setQueryString(queryString) // query everything
+                // startTime: endTs - lookback (conversion)
+                .setStartTimeEpochMs(
+                    1234567) // [Q] double check that these correspond to lookback and
+                // endTs not
+                // min and max Duration
+                .setEndTimeEpochMs(1234567)
+                // [Q] difference between howmany and bucketcount?
+                .setHowMany(10)
+                .setBucketCount(0)
+                .build());
+    List<LogMessage> messages = searchResultToLogMessage(searchResult);
+    List<String> messageStrings = new ArrayList<>();
+    for (LogMessage message : messages) {
+      Map<String, Object> source = message.getSource();
+      final String messageTraceId = (String) source.get("trace_id");
+      final String messageId = message.id;
+      final String messageParentId = (String) source.get("parent_id");
+      // [Q]timestamp not completely correct
+      final String messageTimestamp = (String) source.get("@timestamp");
+      Instant instant = Instant.parse(messageTimestamp);
+      final long messageTimestampMicros =
+          TimeUnit.SECONDS.toMicros(instant.getEpochSecond())
+              + TimeUnit.NANOSECONDS.toMicros(instant.getNano());
+      final long messageDurationMicros = ((Number) source.get("duration_ms")).longValue();
+      final String messageServiceName = (String) source.get("service_name");
+      final String messageName = (String) source.get("name");
+      // [Q]what to put for msgtype
+      final String messageMsgType = "test message type";
```

Comment:
I would just change the `SpanUtil#makeSpanBuilder` class to add a null check around the  msgType and then here pass `null` 

```
    if (msgType != null) {
      tags.add(
              Trace.KeyValue.newBuilder()
                      .setKey(LogMessage.SystemField.TYPE.fieldName)
                      .setVTypeValue(Trace.ValueType.STRING.getNumber())
                      .setVStr(msgType)
                      .build());
    }
```